api/api.go

```diff
@@ -55,3 +57,38 @@ type Config struct {
 	Hosts    []string `yaml:"hosts"`
 	Keyspace string   `yaml:"keyspace"`
 }
+
+// ProfilingAPI wraps an ordinary API and also records profiling metrics to a given Profiler object.
+type ProfilingAPI struct {
```

Comment:
actually, nvm, `backend` has reference to `profiler`, which would not work currently.